Sources/IR/Function.swift

```diff
@@ -88,4 +103,369 @@ public struct Function {
     return result
   }
 
+
+  /// Appends to `self` a basic block in `scope` that accepts `parameters`, returning its address.
+  ///
+  /// The new block will become the function's entry if `self` contains no block before
+  /// `appendBlock` is called.
+  ///
+  /// TODO: merge `appendBlock` and `appendEntry`
+  mutating func appendBlock<T: ScopeID>(
+    in scope: T, taking parameters: [IR.`Type`]
+  ) -> Block.ID {
+    Block.ID(blocks.append(Block(scope: AnyScopeID(scope), inputs: parameters)))
+  }
+
+  /// Appends to `self` an entry block that is in `scope`, returning its identifier.
+  @discardableResult
+  mutating func appendEntry<T: ScopeID>(in scope: T) -> Block.ID {
+    assert(blocks.isEmpty)
+
+    // In functions, the last parameter of the entry denotes the function's return value.
+    var parameters = inputs.map({ `Type`.address($0.type.bareType) })
+    if !isSubscript {
+      parameters.append(.address(output))
+    }
+
+    return appendBlock(in: scope, taking: parameters)
+  }
+
+  /// Removes `block` and updates def-use chains.
+  ///
+  /// - Requires: No instruction in `block` is used by an instruction outside of `block`.
+  @discardableResult
+  mutating func removeBlock(_ block: Block.ID) -> Block {
+    for i in self.instructions(in: block) {
+      precondition(self.allUses(of: i).allSatisfy({ $0.user.block == block.address }))
+      removeUsesMadeBy(i)
+    }
+    return blocks.remove(at: block.address)
+  }
+
+  /// Replaces `old` for `new`.
+  ///
+  /// `old` is removed and the def-use chains are updated so that the uses made by `old` are
+  /// replaced by the uses made by `new` and all uses of `old` refer to `new`. After the call,
```

Comment:
```suggestion
  /// replaced by the uses made by `new` and all former uses of `old` refer to `new`. After the call,
```
 
Ideally this should just be captured in an invariant of `Function` that is documented in its description (or even implied by the meaning of def-use chain!), though that [might require making some variables not publicly settable.](https://val-qs97696.slack.com/archives/C01MEFN3TGE/p1761233302537269)